src/bin/stratis-utils.rs

```diff
@@ -43,6 +116,153 @@ fn base32_decode(var_name: &str, base32_str: &str) -> Result<(), Box<dyn Error>>
     Ok(())
 }
 
+// Get a prediction of filesystem size given a list of filesystem sizes and
+// whether or not the pool allows overprovisioning.
+fn get_filesystem_prediction(
+    overprovisioned: bool,
+    filesystem_sizes: Vec<Bytes>,
+) -> Result<Sectors, Box<dyn Error>> {
+    filesystem_sizes
+        .iter()
+        .map(|&val| {
+            if !(FS_LOGICAL_SIZE_MIN..FS_LOGICAL_SIZE_MAX).contains(&val) {
```

Comment:
I'm not positive but unless there's an optimization specifically for this case, I think this iterates through every single value in between these two very far apart numbers. I think this may actually cause performance problems on the higher end of the `FS_LOGICAL_SIZE_MAX` side. Comparison operators seems more suitable here because they're constant time.

Reply:
clippy made me do it..,I've determined that it is really quite efficient, and compares against the start and end values in the expected manner. Best to leave as is, I think.

Reply:
Algorithmically, the code just compares against the endpoints. What the GitHub issue is about is a failure of a peephole optimization. The comparisons to the two endpoints are made, but somehow they're not simplified as much as the person who filed the issue correctly thinks they should be in some older versions of Rust. Something for the LLVM developer to be concerned about, but not us.